### PR TITLE
Enable async-acks for tracing by default

### PR DESCRIPTION
**Pull request by arajkumar**
## Description

Tracing ingestion pipeline in jaeger doesn't do any retry incase of span write failure, hence we are enabling async ack for tracing by default.

It also adds a flag to turn off streaming span writer incase if needed. By default streaming span writer is enabled.

Fixes https://github.com/timescale/promscale/issues/1632
Fixes https://github.com/timescale/promscale/issues/1635

## Merge requirements

Please take into account the following non-code changes that you may need to make with your PR:

- [x] CHANGELOG entry for user-facing changes
- [ ] Updated the relevant documentation

### CODE REVIEW

File: pkg/jaeger/store/store.go [hunk 1]
```diff
@@ -45,7 +49,17 @@ func (p *Store) SpanWriter() spanstore.Writer {
 }
 
 func (p *Store) StreamingSpanWriter() spanstore.Writer {
-	return p
+	return streamingSpanWriter{p}
+}
+
+func (s streamingSpanWriter) WriteSpan(ctx context.Context, span *model.Span) error {
+	go func() {
```
Comment (alejandrodnm) [inline pkg/jaeger/store/store.go:56]: Should we add some back-preasure mechanism? Something like putting it behind a queue with a pool of goroutines that call the `WriteSpan`?

Comment (cevian) [reply pkg/jaeger/store/store.go:56]: Yes backpressure is key here. Why isn't it enough to simply turn off async-acks for the streaming stuff

Comment (arajkumar) [reply pkg/jaeger/store/store.go:56]: Sure, lets add a back pressure based on the defaults from https://github.com/jaegertracing/jaeger/blob/c1bb2946e670129314bf88d9730d8ed7566766d4/plugin/storage/grpc/shared/streaming_writer.go#L32.

Comment (arajkumar) [reply pkg/jaeger/store/store.go:56]: Looks like I missed the point that the existing batching implementation already has the [back-pressure mechanism](https://github.com/timescale/promscale/blob/efec5ae94de49b1afcca4d3abeff13b8b14b2c59/pkg/pgmodel/ingestor/trace/trace_dispatcher.go#L63) and I can completely skip creating new go routine and turn on async ack as mentioned by @cevian .

**Review by alejandrodnm — APPROVED** [review]

**Review by cevian — APPROVED** [review]

**Pull request merged**